Storage/SQL/SQLitePasswords.swift

```diff
@@ -67,16 +67,18 @@ private class PasswordsTable<T>: GenericTable<Password> {
 
     override var factory: ((row: SDRow) -> Password)? {
         return { row -> Password in
-            let site = Site(url: row["hostname"] as String, title: "")
-            let pw = Password(site: site, username: row["username"] as String, password: row["password"] as String)
-            pw.httpRealm = row["httpRealm"] as String
-            pw.formSubmitUrl = row["formSubmitUrl"] as String
-            pw.usernameField = row["usernameField"] as String
-            pw.passwordField = row["passwordField"] as String
+            let site = Site(url: row["hostname"] as? String ?? "", title: "")
+            let pw = Password(site: site, username: row["username"] as? String ?? "", password: row["password"] as? String ?? "")
+
+            pw.httpRealm = row["httpRealm"] as? String ?? ""
+            pw.formSubmitUrl = row["formSubmitUrl"] as? String ?? ""
+            pw.usernameField = row["usernameField"] as? String ?? ""
+            pw.passwordField = row["passwordField"] as? String ?? ""
             pw.guid = row["guid"] as? String
-            pw.timeCreated = NSDate(timeIntervalSince1970: row["timeCreated"] as Double)
-            pw.timeLastUsed = NSDate(timeIntervalSince1970: row["timeLastUsed"] as Double)
-            pw.timePasswordChanged = NSDate(timeIntervalSince1970: row["timePasswordChanged"] as Double)
+            pw.timeCreated = NSDate(timeIntervalSince1970: row["timeCreated"] as? Double ?? 0)
+            pw.timeLastUsed = NSDate(timeIntervalSince1970: row["timeLastUsed"] as? Double ?? 0)
+            pw.timePasswordChanged = NSDate(timeIntervalSince1970: row["timePasswordChanged"] as? Double ?? 0)
```

Comment:
In short: because we've seen malformed password records in the past, and I would rather fall back to a sane value here than crash the browser. sqlite doesn't stop crazy shit like storing strings in numeric columns, and there's always the chance that some version of Sync, at some point in the future, will fuck up and put shit in the DB.

In other words: yes, it will paper over bugs, but I'm not convinced that a crash here will make the bug any easier to find.